app/mm-redux/utils/post_utils.ts

```diff
@@ -78,9 +77,13 @@ export function canDeletePost(state: GlobalState, config: any, license: any, tea
 
     // Backwards compatibility with pre-advanced permissions config settings.
     if (license.IsLicensed === 'true') {
-        return (config.RestrictPostDelete === General.PERMISSIONS_ALL && (isOwner || isAdmin)) ||
+        if (!isMinimumServerVersion(serverVersion, 6, 0, 0)) {
```

Comment:
I don't think we need this because `hasNewPermissions` already checks for a minimum server version of 4.9.0